### ISSUE TITLE
No CrashDump when using gcc flag "-Og"

### ISSUE THREAD

**Issue by danielsngrbt** [adamgreen/CrashCatcher#14]
No CrashDump is created when using compiler flag (arm-non-eabi-gcc) "-Og", which is a good flag for debugging. Using other flags like "-O2" or even "-O -g" causes no problems and the dump can be used with CrashDebug.

Do you have any ideas, @adamgreen? Already much thanks for your work and your great tool!

**Comment by danielsngrbt** [comment]
Probably I should add that `CrashCatcher_DumpStart` is called, but I don't get anything from `CrashCatcher_GetMemoryRegions`.

**Comment by adamgreen** [comment]
Would is be possible for you to send me a .elf which has this problem? I would like to see the code gen.

**Comment by danielsngrbt** [comment]
Unfortunately I'm not allowed to share the .elf.

**Comment by adamgreen** [comment]
If you build CrashCatcher.c with -O2 and the rest of your code with -Og, does the problem go away? 

Can you attach a JTAG/SWD debugger when it gets into this non-dumping state to see if it is hanging somewhere? If it is just reseting, can you dump the reset reason to see if it is something like a watchdog firing or double faulting?

**Comment by adamgreen** [comment]
You may also want to try increasing CRASH_CATCHER_STACK_WORD_COUNT in CrashCatcherPriv.h  Compiling with -Og might put a greater demand on the stack used by CrashCatcher.

**Comment by danielsngrbt** [comment]
I tried all of your suggestions. Here a quick summary, maybe that's helpful for you, @adamgreen:

- Increasing `CRASH_CATCHER_STACK_WORD_COUNT` from 125 to 2500 and building with our normal debugging options for gcc did the job for me. I wasn't able to retrieve all of our debug symbols with CrashDebug yet, but that's definitely a good start.
- Building `CrashCatcher.c` with -O2 and the rest of the code with -Og helped a bit, but resulted in an infinite loop. `HardFault_Handler` returned successfully, but was called infinitely. I called `HardFault_Handler` manually and stopped it after the first execution in order to inspect the crash dump bin. In the SWD debugger backtrace I saw that it was called two times, so maybe an actual HardFault occurred right after manually "producing" it.
- Building everything with our debugging options (including `CrashCatcher.c)` and leaving `CRASH_CATCHER_STACK_WORD_COUNT` at 125 resulted in a - what I think - memory fault. The CrashDump memory region function didn't return because at some point in it the pc register suddenly contained the value 0x00. I guess that the stack size was too small, see above.

If I find out something else, I will post it here or message you, but I think the issue can be closed. Thanks for your help!
